ext/faiss/index.cpp

```diff
@@ -125,13 +137,20 @@ void init_index(Rice::Module& m) {
       })
     .define_method(
       "search",
-      [](faiss::Index &self, numo::SFloat objects, size_t k) {
+      [](Rice::Object wrapper, numo::SFloat objects, size_t k) {
+        const auto& self = *Rice::Data_Object<faiss::Index>{wrapper};
         auto n = check_shape(objects, self.d);
 
         auto distances = numo::SFloat({n, k});
         auto labels = numo::Int64({n, k});
 
-        self.search(n, objects.read_ptr(), k, distances.write_ptr(), labels.write_ptr());
+        if (wrapper.is_frozen()) {
+          without_gvl([&] {
```

Comment:
This would be helfpul to add to Rice. Pybind does it like this:

https://pybind11.readthedocs.io/en/stable/advanced/misc.html#global-interpreter-lock-gil